Source/FolioReaderCenter.swift

```diff
@@ -387,6 +389,35 @@ open class FolioReaderCenter: UIViewController, UICollectionViewDelegate, UIColl
         
         navigationController?.present(highlighterMenuViewController, animated: true, completion: nil)
     }
+    
+    // MARK: - Shape Options -
+    private func showShapeOptions(from button: UIButton) {
+        guard let shapeMenuViewController = UIStoryboard(name: "ShapeMenu", bundle: Bundle(for: ShapeMenuViewController.self)).instantiateInitialViewController() as? ShapeMenuViewController else { return }
+        
+        addCanvasView(with: .shape)
+        
+        shapeMenuViewController.modalPresentationStyle = .popover
+        shapeMenuViewController.preferredContentSize = CGSize(width: 260, height: 425)
+        shapeMenuViewController.popoverPresentationController?.permittedArrowDirections = .any
+        shapeMenuViewController.popoverPresentationController?.sourceView = button
+        
+        shapeMenuViewController.popoverPresentationController?.sourceRect = CGRect(x: button.bounds.midX, y: button.bounds.minY, width: 0, height: 0)
+        
+        shapeMenuViewController.createShape = { [weak self] viewModel in
```

Comment:
Personally I've decided to use [weak] over [unowned] since I've experienced some weird cases on other apps where self is nil and the app crashes because unowned.